compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -456,8 +456,10 @@ impl<'f> PerFunctionContext<'f> {
 
                 // If there was another store to this instruction without any (unremoved) loads or
                 // function calls in-between, we can remove the previous store.
-                if let Some(last_store) = references.last_stores.get(&address) {
-                    self.instructions_to_remove.insert(*last_store);
+                if !self.aliased_references.contains_key(&address) {
```

Comment:
So this keeps the store even if all instructions which end up using its effect are removed, right? Maybe there is a way to track that after `store v1 at v3` we have `v3` as an alias of `v1`, and thus `v9 = load v3` is using the value in `store v8 at v1`? So that we can somehow decide later whether to remove the last store in `is_store_alias_used`?